lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -10494,20 +10550,27 @@ namespace Js
         return JavascriptOperators::GetTypeId(instance) == TypeIds_Undefined;
     }
 
-    BOOL JavascriptOperators::IsUndefinedObject(Var instance, RecyclableObject *libraryUndefined)
+    BOOL JavascriptOperators::IsUndefinedObject(RecyclableObject* instance)
     {
-        Assert(JavascriptOperators::IsUndefinedObject(libraryUndefined));
+        return instance->GetType()->GetTypeId() == TypeIds_Undefined;
+    }
 
+    BOOL JavascriptOperators::IsUndefinedObject(Var instance, RecyclableObject* libraryUndefined)
+    {
+        Assert(JavascriptOperators::IsUndefinedObject(libraryUndefined));
+        AssertMsg((instance == libraryUndefined)
+          == JavascriptOperators::IsUndefinedObject(instance), "Wrong ScriptContext?");
         return instance == libraryUndefined;
     }
 
     BOOL JavascriptOperators::IsUndefinedObject(Var instance, ScriptContext *scriptContext)
     {
-        return JavascriptOperators::IsUndefinedObject(instance, scriptContext->GetLibrary()->GetUndefined());
+        return JavascriptOperators::IsUndefinedObject(instance, scriptContext->GetLibrary());
     }
 
     BOOL JavascriptOperators::IsUndefinedObject(Var instance, JavascriptLibrary* library)
     {
+        Assert(!RecyclableObject::Is(instance) ? TRUE : ((RecyclableObject*)instance)->GetScriptContext()->GetLibrary() == library );
```

Comment:
From what I understand, this Assert is wrong in Cross Context scenario.
That wasm test just happen to exercice that code path, but it's not a wasm related bug.